sqlalchemy_redshift/dialect.py

```diff
@@ -301,6 +307,11 @@ def _fetch_redshift_column_attributes(self, column):
         return text
 
 
+class RedshiftExecutionCntext(PGExecutionContext_psycopg2):
+    def should_autocommit_text(self, statement):
```

Comment:
That would indeed be a more robust solution, allowing us to automatically stay compatible with future changes to the psycopg2 dialect.